### PR TITLE
Feature/add file encoding

### PR DESCRIPTION
**Pull request by x1mx1x**

### CODE REVIEW

File: CharlieBackend.Api/Controllers/ImportController.cs [hunk 1]
```diff
@@ -17,29 +18,66 @@ namespace CharlieBackend.Api.Controllers
...
+            else
+            {
+                return BadRequest(new { message = "Invalid file extension" });
+            }
+
+            return listOfImportedStudents.ToActionResult();
+        }
     }
-}
+}
```
Comment (SnipeFrost) [inline CharlieBackend.Api/Controllers/ImportController.cs:83]: add new line in the end of file

File: CharlieBackend.Business/Services/GroupImportService.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+        public async Task<Result<List<StudentGroupFileModel>>> ImportFileAsync(IFormFile uploadedFile)
+        {
+            string path = "";
+            List<StudentGroupFileModel> importedGroups = new List<StudentGroupFileModel>();
+
+            if (uploadedFile != null)
+            {
+                path = await CreateFile(uploadedFile);
+                var wb = new XLWorkbook(path);
```
Comment (SnipeFrost) [inline CharlieBackend.Business/Services/GroupImportService.cs:39]: what is it wb, maybe need change name for more readable

File: CharlieBackend.Business/Services/GroupImportService.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                var wsGroups = wb.Worksheet("Groups");
+
+                Type sgType = typeof(StudentGroupFileModel);
```
Comment (SnipeFrost) [inline CharlieBackend.Business/Services/GroupImportService.cs:42]: Same as above about "wsGroups" and "sgType"

File: CharlieBackend.Business/Services/StudentImportService.cs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+            List<StudentFileModel> importedAccounts = new List<StudentFileModel>();
+            var groupName = _unitOfWork.StudentGroupRepository.SearchStudentGroup(groupId).Name;
+            if (!await _unitOfWork.StudentGroupRepository.IsGroupNameExistAsync(groupName))
+            {
+                return Result<List<StudentFileModel>>.GetError(ErrorCode.NotFound, $"Group {groupName} doesn't exist.");
+            }
+
+            if (uploadedFile != null)
+            {
+                var wb = new XLWorkbook(await CreateFile(uploadedFile));
```
Comment (SnipeFrost) [inline CharlieBackend.Business/Services/StudentImportService.cs:53]: Same as above here

**Review by SnipeFrost — CHANGES_REQUESTED** [review]

File: CharlieBackend.Data/Repositories/Impl/StudentRepository.cs [hunk 5]
```diff
@@ -15,6 +16,8 @@ public StudentRepository(ApplicationContext applicationContext)
         {
         }
 
+
```
Comment (SerjKocherga) [inline CharlieBackend.Data/Repositories/Impl/StudentRepository.cs:19]: Remove these empty lines.

File: CharlieBackend.Core/FileModels/StudentFileModel.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (SerjKocherga) [inline CharlieBackend.Core/FileModels/StudentFileModel.cs:2]: Fix using placement.

File: CharlieBackend.Business/Services/Interfaces/IStudentGroupService.cs [hunk 7]
```diff
@@ -1,4 +1,5 @@
 ﻿using CharlieBackend.Core.DTO.StudentGroups;
+using CharlieBackend.Core.Entities;
```
Comment (SerjKocherga) [inline CharlieBackend.Business/Services/Interfaces/IStudentGroupService.cs:2]: Fix using placement for code convention.

**Review by SerjKocherga — CHANGES_REQUESTED** [review]

**Review by BohdanBBG — APPROVED** [review]

**Review by SnipeFrost — APPROVED** [review]

**Review by Hlib-mats — APPROVED** [review]

**Review by SerjKocherga — APPROVED** [review]

**Review by filinskyi — APPROVED** [review]

**Pull request merged**